### PR TITLE
Update containerd related tests using latest kubekins image

### PR DESCRIPTION
**Pull request by bsdnet**
Signed-off-by: Roy Yang <royyang@google.com>

What happened:

The following tests failed continuously in the past:
https://testgrid.k8s.io/sig-node-containerd#containerd-e2e-gci-1.2
https://testgrid.k8s.io/sig-node-containerd#containerd-node-e2e-1.2
https://testgrid.k8s.io/sig-node-containerd#containerd-node-e2e-features-1.2

What you expected to happen:
They should succeed.

How to reproduce it (as minimally and precisely as possible):
See https://testgrid.k8s.io/sig-node-containerd

Please provide links to example occurrences, if any:
See https://testgrid.k8s.io/sig-node-containerd

Anything else we need to know?:
From the log of https://testgrid.k8s.io/sig-node-containerd#containerd-e2e-gci-1.2, it mentioned that failing to get k8s image 1.3, which is removed from https://github.com/kubernetes/test-infra/blob/master/images/kubekins-e2e/variants.yaml. Also, kubekins container is out of date.

Ideally, we should use the latest stable kubekins image. Also, the reason bumping to 1.15 is that it is using the same golang version with latest containered.

### CODE REVIEW

**Comment by MHBauer** [comment]
should these be updated as part of https://github.com/kubernetes/test-infra/pull/17715 ? how does one get into that automated flow?

**Comment by bsdnet** [comment]
> should these be updated as part of #17715 ? how does one get into that automated flow?

That is a good question. I am not sure how other images are updated automatically. From the code reading, this image is special, and it use given verison golang and kubernetes. Recently, 1.3 was removed
from https://github.com/kubernetes/test-infra/blame/master/images/kubekins-e2e/variants.yaml

**Comment by spiffxp** [comment]
>how does one get into that automated flow?

Use an image that gets built and pushed automatically.  In this case, use a version of kubekins that lines up with a version of kubernetes that is still built and supported.  

According the [current documented policy](https://kubernetes.io/docs/setup/release/version-skew-policy/#supported-versions), 1.15 is no longer in support.  You'll want to move to 1.16 to keep up.  In practice, the aged-out release jobs are kept around for a few weeks to cut a final patch release.  Previous release schedules have included a date for when the old jobs would be removed, but I don't see one on the [current release schedule](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.19#timeline) (@kubernetes/patch-release-team @onlydole ?)

Longer term, jobs that need a per-release-branch variant should use [`config-forker` annotations](https://github.com/kubernetes/test-infra/tree/master/releng/config-forker#supported-annotations) to have a [job generated when the release-branch is cut](https://github.com/kubernetes/test-infra/tree/master/config/jobs#release-branch-jobs)

The code that does the bumping is here, fwiw: https://github.com/kubernetes/test-infra/tree/master/experiment/autobumper

**Comment by spiffxp** [comment]
/approve
All the above said, I'll merge this to unblock, as it's not making things any worse.  Please address in followup PR's.

**Comment by onlydole** [comment]
Thank you for the heads up, @spiffxp! It looks like the 1.15 removal dates fell off our schedule with our revisions and we’ll get them re-added 👍🏼

**Comment by bsdnet** [comment]
Thanks @spiffxp  for your detailed explanation.

containered 1.2.x(https://github.com/containerd/containerd/releases/tag/v1.2.13)  is officially built with golang 1.12.17 . That is why I bump to 1.15, which is the only version using the same golang per
https://github.com/kubernetes/test-infra/blob/master/images/kubekins-e2e/variants.yaml

I will follow up to move to 1.16, and it will be also nice for auto bumper to do that(remove from variant, bump the version.) I will take a look at that too.

**Comment by onlydole** [comment]
@spiffxp - to follow up on the 1.15 job removal, I synced up with @justaugustus and SIG Release purposely removed the "1.15 job removal" entry from the schedule. The 1.15 jobs don't exactly correlate to any 1.19 workstreams, and previously, the beta.0 branch was a good time to perform that removal.

Now that the branch cut is on RC.0 AND we have an extended cycle, it's a little too late to have that branch around.

TL;DR - @justaugustus will be removing the 1.15 branch jobs as there are some expectations that the config-forker tooling has, so he will need to verify it still works when someone removes branch jobs "earlier" than the tool expects.

Please let me know if that makes sense, or if there are any other questions on that front, and we can address them!
